litmus-portal/graphql-server/utils/misc.go

```diff
@@ -37,20 +46,61 @@ func ManifestParser(id, key, template string, subscriberConfig *types.Subscriber
 	scanner := bufio.NewScanner(file)
 	var lines []string
 
+	if !*cluster.AgentNsExists || !*cluster.AgentSaExists {
+		comment := fmt.Sprintf("# This is an auto-generated file. DO NOT EDIT")
+		lines = append(lines, comment)
+	}
+	if !*cluster.AgentNsExists {
+		var nsYaml string
+		if cluster.AgentNamespace != nil {
+			nsYaml = fmt.Sprintf("apiVersion: v1\nkind: Namespace\nmetadata:\n  name: %s\n---", *cluster.AgentNamespace)
+		} else {
+			nsYaml = fmt.Sprintf("apiVersion: v1\nkind: Namespace\nmetadata:\n  name: litmus\n---")
+		}
+		lines = append(lines, nsYaml)
+	}
+	if !*cluster.AgentSaExists {
+		var saYaml string
+		if cluster.Serviceaccount != nil && cluster.AgentNamespace != nil {
+			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: %s\n  namespace: %s\n---", *cluster.Serviceaccount, *cluster.AgentNamespace)
+		} else if cluster.Serviceaccount == nil && cluster.AgentNamespace != nil {
+			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: litmus\n  namespace: %s\n---", *cluster.AgentNamespace)
+		} else if cluster.Serviceaccount != nil && cluster.AgentNamespace == nil {
+			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: %s\n  namespace: litmus\n---", *cluster.Serviceaccount)
+		} else {
+			saYaml = fmt.Sprintf("apiVersion: v1\nkind: ServiceAccount\nmetadata:\n  name: litmus\n  namespace: litmus\n---")
+		}
+		lines = append(lines, saYaml)
+	}
+
 	for scanner.Scan() {
 		line := scanner.Text()
-		if strings.Contains(line, "#{CID}") {
-			line = strings.Replace(line, "#{CID}", id, -1)
+		if strings.Contains(line, "# This is an auto-generated file. DO NOT EDIT") {
+			if !*cluster.AgentNsExists || !*cluster.AgentSaExists {
+				line = strings.Replace(line, "# This is an auto-generated file. DO NOT EDIT", "", -1)
+			}
+		} else if strings.Contains(line, "#{CID}") {
+			line = strings.Replace(line, "#{CID}", cluster.ClusterID, -1)
 		} else if strings.Contains(line, "#{KEY}") {
-			line = strings.Replace(line, "#{KEY}", key, -1)
+			line = strings.Replace(line, "#{KEY}", cluster.AccessKey, -1)
 		} else if strings.Contains(line, "#{SERVER}") {
 			line = strings.Replace(line, "#{SERVER}", subscriberConfig.GQLServerURI, -1)
 		} else if strings.Contains(line, "#{SUB-IMAGE}") {
 			line = strings.Replace(line, "#{SUB-IMAGE}", subscriberConfig.SubscriberImage, -1)
 		} else if strings.Contains(line, "#{AGENT-NAMESPACE}") {
-			line = strings.Replace(line, "#{AGENT-NAMESPACE}", subscriberConfig.AgentNamespace, -1)
+			if cluster.AgentNamespace != nil {
+				line = strings.Replace(line, "#{AGENT-NAMESPACE}", *cluster.AgentNamespace, -1)
+			} else {
+				line = strings.Replace(line, "#{AGENT-NAMESPACE}", "litmus", -1)
+			}
+		} else if strings.Contains(line, "#{SUBSCRIBER-SERVICE-ACCOUNT}") {
+			if cluster.Serviceaccount != nil {
```

Comment:
```suggestion
			if cluster.Serviceaccount != nil || cluster.Serviceaccount != ""{
```
- nil when service account didn't pass and ""  when service account passed with an empty string

Reply:
Done.